Main/src/main/java/com/main/acad/controller/Controller.java

```diff
@@ -1,37 +1,53 @@
-import com.main.acad.dao.ChapterDao;
-import com.main.acad.dao.Dao;
 import com.main.acad.entity.Chapter;
-import com.main.acad.serializator.MyJsonSerializer;
+import com.main.acad.serializator.JsonSerializatorImplementation;
+import com.main.acad.service.StrategySimulationServiceImplementation;
 
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 import java.io.IOException;
-import java.io.PrintWriter;
+import java.sql.SQLException;
 import java.util.List;
+import java.util.logging.Logger;
 
 public class Controller {
+    private static final Logger logger = Logger.getLogger(Controller.class.getName());
+
+    StrategySimulationServiceImplementation strategySimulationServiceImplementation = StrategySimulationServiceImplementation.getInstance();
+
     @MappingMethod(url = "api/chapter")
-    public HttpServletResponse returnChapters(HttpServletRequest request, HttpServletResponse response)
-            throws ClassNotFoundException, IllegalAccessException, IOException {
-        Dao dao = new ChapterDao();
-        List chaptersList = dao.listChapters();
-        MyJsonSerializer m = new MyJsonSerializer();
-        String result = m.write(chaptersList);
-        response.getWriter().write(result);
-        return response;
+    public void returnChapters(HttpServletRequest request, HttpServletResponse response) {
+        try {
+            List chaptersList = strategySimulationServiceImplementation.getlistChapters();
+            JsonSerializatorImplementation jsonSerializator = new JsonSerializatorImplementation();
+            String result = jsonSerializator.write(chaptersList);
+            response.getWriter().write(result);
+        } catch (IllegalAccessException | IOException e) {
+            logger.info("An error occurred in the Controller in the returnChapters method");
+        }
     }
 
     @MappingMethod(url = "api/subChapter")
-    public static HttpServletResponse returnSubChapters(HttpServletRequest request, HttpServletResponse response)
-            throws ClassNotFoundException, IllegalAccessException, IOException {
-        Dao d = new ChapterDao();
-        List<String> child = d.listChildren(Integer.parseInt("3"));
-        MyJsonSerializer m2 = new MyJsonSerializer();
-        String result = m2.write(child);
-        response.getWriter().write(result);
-        return response;
+    public void returnSubChapters(HttpServletRequest request, HttpServletResponse response) {
+        List<Chapter> child = strategySimulationServiceImplementation.getlistChildren(Integer.parseInt(request.getParameter("id")));
+        JsonSerializatorImplementation jsonSerializer = new JsonSerializatorImplementation();
+        try {
+            String result = jsonSerializer.write(child);
+            response.getWriter().write(result);
+        } catch (IllegalAccessException | IOException e) {
+            logger.info("An error occurred in the Controller in the returnSubChapters method");
```

Comment:
same

Reply:
I add exception handling

Reply:
I don't see) Printing stacktrace is not handling and should not be used at all (you may use logging). Handling is a way in which user can understand what wrong happens and how to fix it in some cases